### PR TITLE
Front_End: Conventions & Best Practices

### LINKED ISSUE #220: Front_End: Conventions & Best Practices
I would like to write a Doc on _**Conventions & Best Practices**_ while writing HTML code/scripts.

* I would be mentioning best and bad practices.
* an explanation with appropriate examples.

Please Assign this to me. Thanks
@2024-SANDHYA

### PR DESCRIPTION
**Pull request by ravireddy07**
<hr>

## Description 📜

Fixes #220

- Added a readme(`html_conventions_practices.md`) on Conventions & Best Practices while writing HTML code/scripts.

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Frontend Dev (HTML, CSS, JS)

<hr>
 
## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by anushka-mishra-code** [comment]
I would be glad to work on this issue.
Name : Anushka Mishra
Batch : 1
Serial No.: 175
Type : Documentation
Approach:

Introduction to HTML , CSS and JavaScript
List all the maximum coventions to follow
and best practices to make an efficient front page

**Comment by ravireddy07** [comment]
> I would be glad to work on this issue.
> Name : Anushka Mishra
> Batch : 1
> Serial No.: 175
> Type : Documentation
> Approach:
> 
> Introduction to HTML , CSS and JavaScript
> List all the maximum coventions to follow
> and best practices to make an efficient front page

You should request in Issues, not PR's.

**Comment by ravireddy07** [comment]
@code-reaper08 review it once

**Review by code-reaper08 — COMMENTED** [review]
Looks cool @ravireddy07 🆒 
Will merge it soon :-)

And also please ping me regarding this in our batch channel. So, that I can keep a note of it.

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+
+-   Bad Practice:
+
+    ```sh
+    <!DOCTYPE Html>
+    ```
+
+    ```sh
+    <!doctype HTML>
+    ```
```
Comment (praveenscience) [inline Web_Development/FrontEnd/HTML/html_conventions_practices.md:42]: These aren't shell. Please use html:


    ```html
    <!DOCTYPE Html>
    ```

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+-   Easier to write and read
+-   Mixing uppercase and lowercase names looks bad
+
+-   Good Practice:
+
+    ```sh
+    <body>
+        <p>Winter of Contributing</p>
+    </body>
+    ```
```
Comment (praveenscience) [inline Web_Development/FrontEnd/HTML/html_conventions_practices.md:61]: These aren't shell. Please use html:


    ```html
    <!DOCTYPE Html>
    ```

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+
+-   Bad Practice:
+
+    ```sh
+    <BODY>
+        <p>Winter of Contributing</p>
+    </BODY>
+    ```
```
Comment (praveenscience) [inline Web_Development/FrontEnd/HTML/html_conventions_practices.md:69]: These aren't shell. Please use html:


    ```html
    <!DOCTYPE Html>
    ```

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+    <img src="gwoc.gif" alt="GWOC" style="width:128px;height:128px">
+    <a href="https://gwoc.girlscript.tech/">Visit GWOC</a>
+    ```
+
+-   Bad Practice:
+
+    ```sh
+    <img SRC="gwoc.gif" ALT="GWOC" STYLE="width:128px;height:128px">
+    <a HREF="https://gwoc.girlscript.tech/">Visit GWOC</a>
+    ```
```
Comment (praveenscience) [inline Web_Development/FrontEnd/HTML/html_conventions_practices.md:91]: These aren't shell. Please use html:


    ```html
    <!DOCTYPE Html>
    ```

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+      <h2>GSSOC</h2>
+        <p>
+          It will happen in March
+        </p>
+      <h2>GWOC</h2>
+        <p>
+          It will happen in September
+        </p>
+    </body>
+    ```
```
Comment (praveenscience) [inline Web_Development/FrontEnd/HTML/html_conventions_practices.md:124]: These aren't shell. Please use html:


    ```html
    <!DOCTYPE Html>
    ```

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+    <body class=container>
+      <p>Winter of Contributing</p>
+    </body>
+    ```
+
+    ```sh
+    <body class=body container>
+      <p>Winter of Contributing</p>
+    </body>
+    ```
```
Comment (praveenscience) [inline Web_Development/FrontEnd/HTML/html_conventions_practices.md:153]: These aren't shell. Please use html:


    ```html
    <!DOCTYPE Html>
    ```

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md [hunk 7]
```diff
@@ -0,0 +1,408 @@
...
+
+-   Bad Practice:
+
+    ```sh
+    <body>
+      <p>Girlscript Foundation
+      <p>Winter of Contributing
+    </body>
+    ```
```
Comment (praveenscience) [inline Web_Development/FrontEnd/HTML/html_conventions_practices.md:177]: These aren't shell. Please use html:


    ```html
    <!DOCTYPE Html>
    ```

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md [hunk 8]
```diff
@@ -0,0 +1,408 @@
...
+
+    ```sh
+    <img src="gwoc.gif" alt="GWOC" style="width:128px;height:128px"/>
+    ```
+
+-   Bad Practice:
+
+    ```sh
+    <img SRC="html5.gif">
+    ```
```
Comment (praveenscience) [inline Web_Development/FrontEnd/HTML/html_conventions_practices.md:197]: These aren't shell. Please use html:


    ```html
    <!DOCTYPE Html>
    ```

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md [hunk 9]
```diff
@@ -0,0 +1,408 @@
...
+Like Images,
+
+Always specify the alt attribute for images. This attribute is important if the image for some reason cannot be displayed.
+
+Also, always define the width and height of images.Because the browser can reserve space for the image before loading and sometimes image ratio differs.
+
+-   Good Practice:
+
+    ```sh
+    <img src="gwoc.gif" alt="GWOC" style="width:128px;height:128px"/>
```
Comment (praveenscience) [inline Web_Development/FrontEnd/HTML/html_conventions_practices.md:190]: Avoid suggesting the use of inline styles.

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md [hunk 10]
```diff
@@ -0,0 +1,408 @@
...
+
+    ```sh
+    <ul>
+      <li>Managers</li>
+      <li>Admins</li>
+      <li>Supervisors</li>
+      <li>Mentors</li>
+      <li>Contributors</li>
+    </ul>
+    ```
```
Comment (praveenscience) [inline Web_Development/FrontEnd/HTML/html_conventions_practices.md:266]: These aren't shell. Please use html:


    ```html
    <!DOCTYPE Html>
    ```

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md [hunk 11]
```diff
@@ -0,0 +1,408 @@
...
+
+### Never Skip
+
+As we seen above, A HTML page can be validated without writing or closing a tag.
+
+-   The <title> element is required in HTML.
```
Comment (praveenscience) [inline Web_Development/FrontEnd/HTML/html_conventions_practices.md:272]: The `<title>` needs to be in code:

```suggestion
-   The `<title>` element is required in HTML.
```

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md [hunk 12]
```diff
@@ -0,0 +1,408 @@
...
+-   The contents of a page title is very important for search engine optimization (SEO)!
+-   The page title is used by search engine algorithms to decide the order when listing pages in search results.
+-   Omitting <body> tag can produce errors in older browsers.
+-   Omitting <html> tag and <body> can also crash DOM and XML software.
+-   If we omit <head> tag, Browsers will add all elements before <body>, to a default <head> element.
```
Comment (praveenscience) [inline Web_Development/FrontEnd/HTML/html_conventions_practices.md:277]: HTML tags need to be inside code:

Example, this is a `<tag>`.

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md [hunk 13]
```diff
@@ -0,0 +1,408 @@
...
+### HTML Meta Data
+
+The <meta> tag contains additional information related to the document such as page description, keywords, author, character set, etc.
+
+-   The web browsers use the metadata to display content, and keywords are used by search engines and other web-services.
+-   Best Practice:
+
+    ```sh
+    <meta charset="UTF-8">
+    ```
```
Comment (praveenscience) [inline Web_Development/FrontEnd/HTML/html_conventions_practices.md:304]: These aren't shell. Please use html:


    ```html
    <!DOCTYPE Html>
    ```

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md [hunk 14]
```diff
@@ -0,0 +1,408 @@
...
+-   We should write long CSS rules over multiple lines.
+
+    ```sh
+    body {
+      background-color: black;
+      font-family: "Arial Black";
+      font-size: 16em;
+      color: white;
+    }
+    ```
```
Comment (praveenscience) [inline Web_Development/FrontEnd/HTML/html_conventions_practices.md:352]: These aren't shell. Please use html:


    ```html
    <!DOCTYPE Html>
    ```

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md [hunk 15]
```diff
@@ -0,0 +1,408 @@
...
+      <script>
+        // Paragraph 2 will be overwritten
+        document.getElementById("demo").innerHTML = "HELLO!";
+      </script>
+    </body>
+    ```
+
+    ```sh
+    <script src="app.js">
+    ```
```
Comment (praveenscience) [inline Web_Development/FrontEnd/HTML/html_conventions_practices.md:385]: These aren't shell. Please use html:


    ```html
    <!DOCTYPE Html>
    ```

**Review by praveenscience — CHANGES_REQUESTED** [review]
Great commit but a couple of changes, please update these. 😊

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md (hunk 9 above)
Comment (ravireddy07) [reply Web_Development/FrontEnd/HTML/html_conventions_practices.md:190]: yeah

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md (hunk 15 above)
Comment (ravireddy07) [reply Web_Development/FrontEnd/HTML/html_conventions_practices.md:385]: Updated

File: Web_Development/FrontEnd/HTML/html_conventions_practices.md (hunk 12 above)
Comment (ravireddy07) [reply Web_Development/FrontEnd/HTML/html_conventions_practices.md:277]: Updated

**Review by code-reaper08 — APPROVED** [review]
Looks good to me!

**Review by praveenscience — APPROVED** [review]
👍

**Pull request merged**